postgres/src/main/java/org/jdbi/v3/postgres/HStoreArgumentFactory.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.postgres;
+
+import org.jdbi.v3.core.StatementContext;
+import org.jdbi.v3.core.argument.Argument;
+import org.jdbi.v3.core.argument.ArgumentFactory;
+
+import java.lang.reflect.Type;
+import java.sql.Types;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * An argument factory which binds Java's {@link Map} to Postgres' hstore type.
+ */
+public class HStoreArgumentFactory implements ArgumentFactory {
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Optional<Argument> build(Type type, Object value, StatementContext ctx) {
+        if (value instanceof Map) {
```

Comment:
I gave a little bit thought to this question, and I think we should not limit API to the `Map<String, String>` type. The PostgreSQL JDBC driver supports any maps during inserts, and we are just proxy calls to it. By removing the restriction on the `<String, String>` generic parameter we support all types of insertions in JDBI and provide a good developer experience.